AsaApi/Core/Private/Tools/Requests.cpp

```diff
@@ -234,11 +269,59 @@ namespace API
 		return true;
 	}
 
+	bool Requests::CreatePostRequest(const std::string& url, const std::function<void(bool, std::string, std::unordered_map<std::string, std::string>)>& callback,
+		const std::string& post_data, std::vector<std::string> headers, long connectionTimeout, long receiveTimeout, long sendTimeout)
+	{
+		std::thread([this, url, callback, post_data, headers, connectionTimeout, receiveTimeout, sendTimeout]
+			{
+				std::string Result = "";
+				std::unordered_map<std::string, std::string> responseHeaders;
+				Poco::Net::HTTPResponse response(Poco::Net::HTTPResponse::HTTP_BAD_REQUEST);
+				Poco::Net::HTTPClientSession* session = nullptr;
+
+				try
+				{
+					Poco::Net::HTTPRequest&& request = pimpl->ConstructRequest(url, session, headers, Poco::Net::HTTPRequest::HTTP_POST, connectionTimeout, receiveTimeout, sendTimeout);
+
+					request.setContentType("application/x-www-form-urlencoded");
+					request.setContentLength(post_data.length());
+
+					std::ostream& OutputStream = session->sendRequest(request);
```

Comment:
It feels like 95% of the point of the thread is to avoid blocking on the `session->SendRequest(request)` bit...perhaps we could encapsulate just that?